sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -113,6 +113,18 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
       }
     }
 
+    if (requiredSchema.length == 1 &&
+      requiredSchema.head.name == parsedOptions.columnNameOfCorruptRecord) {
+      throw new AnalysisException(
+        s"""
+           |'${parsedOptions.columnNameOfCorruptRecord}' cannot be selected alone without other
```

Comment:
I would explain why in short too, for example, `... selecting this alone does not actually parse other fields ...`.